### PR TITLE
feat: create ORA "date_config_type" field

### PR DESCRIPTION
**Pull request by leangseu-edx**
**TL;DR -** add no_dates field to xblock and let ora use course start, end date.

JIRA: https://2u-internal.atlassian.net/browse/AU-1314

**What changed?**

- Create a new field on the ORA xblock `date_config_type`
- This field set to `manual`, `subsection` and `course_end`

NOTE: this only work with lms. `runtime` libs does not seem to work in studio. `is_course_staff` also not working on studio, so I assume this is acceptable.

**Developer Checklist**

- [ ] Reviewed the [release process](https://github.com/openedx/edx-ora2/blob/master/.github/release_process.md)
- [ ] Translations and JS/SASS compiled
- [ ] Bumped version number in [openassessment/\_\_init\_\_.py](https://github.com/openedx/edx-ora2/blob/master/openassessment/__init__.py#L4) and [package.json](https://github.com/openedx/edx-ora2/blob/master/package.json#L3)

**Testing Instructions**

[ How should a reviewer test this PR? ]

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

**Review by jansenk — CHANGES_REQUESTED** [review]
This is entirely my fault - I forgot to hit "save" on this ticket when I was changing the description. I have gone in and updated the description to the changes we actually want. This is a perfectly valid implementation of the ticket as it was handed to you but isn't actually what we want to do. I take full responsibility for this mistake.

 https://2u-internal.atlassian.net/browse/AU-1314

File: openassessment/xblock/openassessmentblock.py [hunk 1]
```diff
@@ -1127,6 +1133,13 @@ def is_closed(self, step=None, course_staff=None):
         if course_staff:
             return False, None, DISTANT_PAST, DISTANT_FUTURE
 
+        if self.date_config_type == 'course_end':
+            course = self.get_course()
+            if course.start and course.end:
```
Comment (jansenk) [inline openassessment/xblock/openassessmentblock.py:1138]: is it possible for these to not be truthy? what is this checking for?

**Review by jansenk — CHANGES_REQUESTED** [review]
Looks good but a few suggestions

File: openassessment/xblock/openassessmentblock.py (hunk 1 above)
Comment (leangseu-edx) [reply openassessment/xblock/openassessmentblock.py:1138]: Locally, when the course just created. Start or end date doesn't get define. This is for the safety check.

**Review by jansenk — APPROVED** [review]
